test/Microsoft.AspNet.Mvc.Core.Test/ParameterBinding/ModelBindingHelperTest.cs

```diff
@@ -696,6 +696,107 @@ public void GetPropertyName_FieldExpression_Throws()
             Assert.Equal(expectedMessage, exception.Message);
         }
 
+        [Fact]
+        public void ClearValidationStateForModel_EmtpyModelKey()
+        {
+            // Arrange
+            var metadataProvider = new EmptyModelMetadataProvider();
+            ModelStateDictionary dictionary = new ModelStateDictionary();
+            dictionary["MyProperty"] = new ModelState { ValidationState = ModelValidationState.Invalid };
+            dictionary.AddModelError("MyProperty", "MyProperty invalid.");
+            dictionary["IncludedProperty"] = new ModelState { ValidationState = ModelValidationState.Invalid };
+            dictionary.AddModelError("IncludedProperty", "Included Property invalid.");
+            dictionary.AddModelError("IncludedProperty", "Included Property required.");
+            dictionary["ExcludedProperty"] = new ModelState { ValidationState = ModelValidationState.Valid };
+
+            // Act
+            ModelBindingHelper.ClearValidationStateForModel(
+                typeof(MyModel),
+                dictionary,
+                metadataProvider,
+                "");
```

Comment:
make this a Theory method which takes in values string.Empty and null. IN both cases the behavior of the api should be same.